mp/src/game/client/momentum/ui/HUD/hud_stickybombs.cpp

```diff
@@ -1,20 +1,29 @@
 #include "cbase.h"
 
 #include <vgui_controls/EditablePanel.h>
-#include <vgui_controls/Label.h>
+#include <vgui_controls/AnimationController.h>
+#include <vgui_controls/Button.h>
 
 #include "c_mom_player.h"
 #include "hudelement.h"
 #include "iclientmode.h"
 #include "mom_system_gamemode.h"
 #include "weapon/weapon_mom_stickybomblauncher.h"
+#include "weapon/weapon_shareddefs.h"
 
 #include "tier0/memdbgon.h"
 
 using namespace vgui;
 
 static MAKE_TOGGLE_CONVAR(mom_hud_sj_stickycount_enable, "1", FCVAR_ARCHIVE, "Toggles the stickybomb counter.\n");
-static MAKE_TOGGLE_CONVAR(mom_hud_sj_stickycount_autohide, "0", FCVAR_ARCHIVE, "Toggles automatically hiding the stickybomb counter at 0 stickies. 0 = OFF, 1 = ON\n");
+
+struct Stickybox_t
+{
+    Button *button;
+    Color *color;
+    const char *animationSeq;
+    bool flag;
+};
```

Comment:
This is a step in the right direction but this calls for a `StickyBox` vgui control. It doesn't seem to need to be a button, either. Buttons are inherently clickable, this is just a box that has its bg and fg colors animated.

The animation sequence would be the same, you would just be running it explicitly on the StickyBox panel itself.

Reply:
Yeah I was using button as it deals with the border bg coloring but doesn't make much sense here. 
I understand a control for the animation and color sake, but not sure where to keep the flag I use. I'll stuff it in the control for now.

Reply:
Yes the goal is to convert `StickyBox` to be an actual control that has its own colors. Can "flag" be renamed? What does it even mean?

Reply:
For the flag:
- `true` signifies that the sticky is new and can prearm (set prearm color and start the fade-in animation).
- `false` signifies that the stickybox has prearmed (started the animation) and can be reset to `NoStickyColor` when it is det.

So maybe a better name is `seen` or `processed`. 

It is tied to the control yes but it still needs to be accessed by the parent `hud_stickybombs` as the animations are swapped when a sticky is det. This is to keep their current animation when they are moved leftward.
This animation swapping i think is the only thing that really needs to be in the parent, as far as `OnThink` is concerned.